File: loan_calculator.kt
fun main() {
    /*
        Last name: Punsalan
        Language: Kotlin
        Paradigm(s): 
    */

    // reading input
    print("Loan Amount: ")
    var loanAmount = readln()
                    .replace("PHP", "") // removing any "PHP" 
                    .replace(",", "") // removing any commas
                    .trim() // removing any trailing spaces
                    .toFloat() // converting to float
    print("Annual Interest Rate: ")
    var interestRate = readln()
                        .replace("%", "") // removing any percentage signs
                        .trim() // removing any trailing spaces
                        .toFloat() // converting to float
                        interestRate /= 100 // converting percentage to decimal
    print("Loan Term: ")
    var loanTermsYear = readln()
                        .replace("years", "") // removing any "years"
                        .trim() // removing any trailing spaces
                        .toInt() // converting to int

    // calculating values
    var monthlyInterestRate = interestRate / 12
    var loanTermsMonth = loanTermsYear * 12
    var totalInterest = loanAmount * monthlyInterestRate * loanTermsMonth
    var monthlyRepayment = (loanAmount + totalInterest) / loanTermsMonth
    
    // printing output
    println()
    println("Loan Amount: PHP %.2f".format(loanAmount))
    println("Annual Interest Rate: %.2f%%".format(interestRate * 100))
    println("Loan Term: $loanTermsMonth months")
    println("Monthly Repayment: PHP %.2f".format(monthlyRepayment))
    println("Total Interest: PHP %.2f".format(totalInterest))
}

File: loan_calculator.go
package main

/*

Last name: Go
Language: Go
Paradigm(s): Procedural Programming

*/
import (
	"fmt"
)

func main() {
	// Input: Loan amount, annual interest rate, loan term in years
	var loanAmount float64
	var annualInterestRate float64
	var loanTermYears int

	fmt.Print("Enter Loan Amount (PHP): ")
	fmt.Scan(&loanAmount)
	fmt.Print("Enter Annual Interest Rate (%): ")
	fmt.Scan(&annualInterestRate)
	fmt.Print("Enter Loan Term (years): ")
	fmt.Scan(&loanTermYears)

	// Calculate monthly interest rate and loan term in months
	monthlyInterestRate := annualInterestRate / 12 / 100 //divide by 100 to turn decimal
	loanTermMonths := loanTermYears * 12

	// Calculate total interest and monthly repayment
	totalInterest := loanAmount * monthlyInterestRate * float64(loanTermMonths) //go doesn't allow implicit type connversions, so type casting is required  (loanTermMonths will end up as int)
	monthlyRepayment := (loanAmount + totalInterest) / float64(loanTermMonths)

	// Output the results
	fmt.Println("\nLoan Details:")
	fmt.Printf("Loan Amount: PHP %.2f\n", loanAmount)
	fmt.Printf("Annual Interest Rate: %.2f%%\n", annualInterestRate)
	fmt.Printf("Loan Term: %d months\n", loanTermMonths)
	fmt.Printf("Monthly Repayment: PHP %.2f\n", monthlyRepayment)
	fmt.Printf("Total Interest: PHP %.2f\n", totalInterest)
}
